File: snail-race-bet/typescript/src/adapters/MongoDbBetRepository.ts
import {Db} from "mongodb";
import {Bet, BetRepository, PodiumPronostic} from "../domain/BetRepository";

export class MongoDbBetRepository implements BetRepository {
    constructor(private db: Db) {
    }

    async register(bet: Bet): Promise<void> {
        await this.getCollection().insertOne({...bet});
    }
    async findByDateRange(from: number, to: number): Promise<Bet[]> {
        return this.getCollection().find({
            timestamp: {
                $gte: from,
                $lt: to
            }
        }, ).map((d) => new Bet(d.gambler, d.pronostic, d.timestamp)).toArray()
    }

    private getCollection() {
        return this.db.collection('bets');
    }
}

File: snail-race-bet/java/src/main/java/snail/race/kata/adapters/MongoDbBetRepository.java
package snail.race.kata.adapters;

import com.mongodb.client.MongoDatabase;
import snail.race.kata.domain.Bet;
import snail.race.kata.domain.BetRepository;

import java.util.List;

public class MongoDbBetRepository implements BetRepository {
    public MongoDbBetRepository(MongoDatabase database) {

    }

    @Override
    public void register(Bet bet) {
    }

    @Override
    public List<Bet> findByDateRange(int from, int to) {
        return null;
    }
}
